registry.rb

```diff
@@ -0,0 +1,90 @@
+# Registry class gathers the metric of the service.
+# It stores all supported type of the metric e.g. Counters, Gauges and Histograms.
+#
+require 'concurrent'
+require_relative 'meters/counter'
+require_relative 'meters/gauge'
+require_relative 'histogram/histogram'
+
+module Registry
+  class MetricsRegistry
+    class DuplicateKeyError < StandardError; end
+
+    def initialize
+      @store = Concurrent::Map.new
+    end
+
+    # Add metric into registry
+    #
+    # @param metric [Metric] metric to add it can be of any type
+    # @return metric object
+    def add(metric)
+      raise TypeError unless metric.respond_to? :name
+
+      key = metric.name

```

Comment:
this is incorrect
see 

https://github.com/wavefrontHQ/wavefront-pyformance/blob/73b7a43699ee3b4a2e21bf251630b473f27f2d08/wavefront_pyformance/tagged_registry.py#L15

https://github.com/wavefrontHQ/go-metrics-wavefront/blob/e86b11d2e81bbf4e48f1fde1fe8ad5e4b8b5e40e/reporting/utils.go#L10
^^ is used here: https://github.com/wavefrontHQ/go-metrics-wavefront/blob/e86b11d2e81bbf4e48f1fde1fe8ad5e4b8b5e40e/reporting/registry.go#L20